### ISSUE TITLE
"Lexical environment is suspended" when using visitEachChild from nightly transformer API (2.3.0-dev)

### ISSUE THREAD

**Issue by fabiandev** [Microsoft/TypeScript#15192]
I'm experiencing issues with `visitEachChild` for e.g. a variable declaration when using the transformer API from the nightly builds.

**TypeScript Version:**  2.3.0-dev.20170414

**Code**

Sample code to transform:

```ts
let a: () => any;
```

Minimal transformer:

```ts
function transformer(context) {
    const visitor = node => {
        return ts.visitEachChild(node, visitor, context);
    };

    return sf => ts.visitNode(sf, visitor);
}
```

@rkirov pointed out, that a "minimal repro is input function `f(a: () => null) {}` with the identity transformer (i.e. just calling visitEachChild with `(n) => n`" (see https://github.com/Microsoft/TypeScript/pull/13940#issuecomment-293762316)

**Expected behavior:**

Should run without errors.

**Actual behavior:**

`Error: Debug Failure. False expression: Lexical environment is suspended.`

I've reproduced the error on runkit: https://runkit.com/fabiandev/58e571bd9383e400145e1245

**Comment by longlho** [comment]
Hi there is there any activity on this issue :(? Thanks!

**Comment by kimamula** [comment]
Shouldn't this be included in TypeScript 2.3.1 milestone?
I am wondering whether the custom transformer feature will be shipped with the stable release of TypeScript 2.3.

**Comment by fabiandev** [comment]
As of build `2.3.0-dev.20170426`, the example on runkit works fine: https://runkit.com/fabiandev/59005befe4d0d60011238738
